plugins/module_utils/sns.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+import copy
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+@AWSRetry.jittered_backoff()
+def _list_topics_with_backoff(client):
+    paginator = client.get_paginator('list_topics')
+    return paginator.paginate().build_full_result()['Topics']
+
+
+@AWSRetry.jittered_backoff(catch_extra_error_codes=['NotFound'])
+def _list_topic_subscriptions_with_backoff(client, topic_arn):
+    paginator = client.get_paginator('list_subscriptions_by_topic')
+    return paginator.paginate(TopicArn=topic_arn).build_full_result()['Subscriptions']
+
+
+@AWSRetry.jittered_backoff(catch_extra_error_codes=['NotFound'])
+def _list_subscriptions_with_backoff(client):
+    paginator = client.get_paginator('list_subscriptions')
+    return paginator.paginate().build_full_result()['Subscriptions']
```

Comment:
Is it worth combining these in a single function with a param input? for example:
https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/cloudfront_info.py#L319

Might make it easier then if any different paginators get used later